src/select-query-parser/result.ts

```diff
@@ -392,21 +428,46 @@ type ProcessEmbeddedResourceResult<
  * @param Spread - The SpreadNode to process.
  */
 type ProcessSpreadNode<
+  ClientOptions extends ClientServerOptions,
   Schema extends GenericSchema,
   Row extends Record<string, unknown>,
   RelationName extends string,
   Relationships extends GenericRelationship[],
   Spread extends Ast.SpreadNode
-> = ProcessNode<Schema, Row, RelationName, Relationships, Spread['target']> extends infer Result
+> = ProcessNode<
+  ClientOptions,
+  Schema,
+  Row,
+  RelationName,
+  Relationships,
+  Spread['target']
+> extends infer Result
   ? Result extends SelectQueryError<infer E>
     ? SelectQueryError<E>
     : ExtractFirstProperty<Result> extends unknown[]
-    ? {
-        [K in Spread['target']['name']]: SelectQueryError<`"${RelationName}" and "${Spread['target']['name']}" do not form a many-to-one or one-to-one relationship spread not possible`>
-      }
+    ? ClientOptions['postgrestVersion'] extends 13 // Spread over an many-to-many relationship, turn all the result fields into arrays
```

Comment:
Q: Will this comment appear on the docs? If so, it would be good to mention the arrays are correlated (they share the same order).

```suggestion
    ? ClientOptions['postgrestVersion'] extends 13 // Spread over an many-to-many relationship, turn all the result fields into correlated arrays
```

Reply:
Not really this comment is only there for devs to understand why we do branch here and the behavior this type cover. Sounds good to mention the correlation regardless 👍